packages/components/src/Menu/Menu.tsx

```diff
@@ -111,34 +118,41 @@ export function Menu({
   useRefocusOnActivator(visible);
   const menuRef = useFocusTrap<HTMLDivElement>(visible);
 
-  const [popperElement, setPopperElement] = useState<HTMLElement | null>(null);
-  const {
-    styles: popperStyles,
-    attributes,
-    state,
-  } = usePopper(popperRef.current, popperElement, {
+  const { refs, floatingStyles, context } = useFloating({
+    open: visible,
+    onOpenChange: setVisible,
     placement: "bottom-start",
     strategy: "fixed",
-    modifiers: [
-      {
-        name: "flip",
-        options: {
-          flipVariations: true,
+    middleware: [
+      offset(MENU_OFFSET),
+      flip({ flipAlignment: false }),
+      size({
+        apply({ availableHeight, elements }) {
+          // The inner element is the scrollable menu that requires the max height
+          const menuElement = elements.floating.querySelector(
+            '[role="menu"]',
+          ) as HTMLElement;
+
+          if (menuElement) {
+            const maxHeight = Math.min(MENU_MAX_HEIGHT, availableHeight);
```

Comment:
on all these size middlewares, I think I wanna add just a bit more space. it's exactly flush against the edge and it feels, claustrophobic.